### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by briehl**
Tests were unable to run due to an installed incompatibility, so I did the following updates:

* pandas to 2.2.3 (latest), which no longer installs an incompatible version of numpy.
* Updated `Pipfile` (and `Pipfile.lock`) to be aligned with requirements.txt
* (not related) added pytest_cache/ to `.gitignore`

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -7,23 +7,24 @@ verify_ssl = true
...
-aiohttp = "==3.7.4"
-pytest = "==6.2.4"
-aiofiles = "==0.3.1"
-pytest-aiohttp = "==0.1.3"
-coverage = "==4.4.2"
-pytest-cov = "==2.5.1"
-hypothesis = "==6.14.5"
-globus-sdk = "==1.6.1"
-python-dotenv = "==0.14.0"
+aiofiles = "==23.1.0"
```
Comment (briehl) [inline Pipfile:10]: Alphabetized these to make it easier to compare with requirements.txt

File: Pipfile [hunk 2]
```diff
@@ -7,23 +7,24 @@ verify_ssl = true
...
+aiohttp = "==3.8.4"
 aiohttp_cors = "==0.7.0"
-frozendict = "==2.0.3"
-pandas = "==1.3.2"
-xlrd = "==2.0.1"
-openpyxl = "==3.0.7"
+black = "==23.7.0"
```
Comment (briehl) [inline Pipfile:13]: Added black, wasn't here before.

Comment (ialarmedalien) [reply Pipfile:13]: might be easier to use ruff -- linting and code formatting in one beautiful package! Can remove flake8, etc.

File: Pipfile (hunk 1 above)
Comment (MrCreosote) [inline Pipfile:10]: There's a jump for you

**Review by MrCreosote — APPROVED** [review]

File: Pipfile (hunk 2 above)
Comment (bio-boris) [reply Pipfile:13]: We have ruff already here  as of https://github.com/kbase/staging_service/pull/196

Comment (briehl) [reply Pipfile:13]: I'm happy to reorganize that, too, if we like. I'll get it in a separate PR. I just wanted the Pipfile and requirements up to date with each other (more or less).

**Pull request merged**